code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,816 @@
+	name = "Prepare Blood Magic"
+	button_icon_state = "carve"
+	desc = "Prepare blood magic by carving runes into your flesh. This is easier with an <b>empowering rune</b>."
+	var/list/spells = list()
+	var/channeling = FALSE
+
+/datum/action/innate/cult/blood_magic/Grant()
+	..()
+	button.ordered = FALSE
+	button.screen_loc = DEFAULT_BLOODSPELLS
+	button.moved = DEFAULT_BLOODSPELLS
+
+/datum/action/innate/cult/blood_magic/Remove()
+	for(var/X in spells)
+		qdel(X)
+	..()
+
+/datum/action/innate/cult/blood_magic/proc/Positioning()
+	var/list/screen_loc_split = splittext(button.screen_loc,",")
+	var/list/screen_loc_X = splittext(screen_loc_split[1],":")
+	var/list/screen_loc_Y = splittext(screen_loc_split[2],":")
+	var/pix_X = text2num(screen_loc_X[2])
+	for(var/datum/action/innate/cult/blood_spell/B in spells)
+		if(B.button.locked)
+			var/order = pix_X + spells.Find(B) * 31
+			B.button.screen_loc = "[screen_loc_X[1]]:[order],[screen_loc_Y[1]]:[screen_loc_Y[2]]"
+			B.button.moved = B.button.screen_loc
+
+/datum/action/innate/cult/blood_magic/Activate()
+	var/rune = FALSE
+	var/limit = RUNELESS_MAX_BLOODCHARGE
+	for(var/obj/effect/rune/empower/R in range(1, owner))
+		rune = TRUE
+		break
+	if(rune)
+		limit = MAX_BLOODCHARGE
+	if(spells.len >= limit)
+		if(rune)
+			to_chat(owner, "<span class='cultitalic'>You cannot store more than [MAX_BLOODCHARGE] spells. <b>Pick a spell to remove.</b></span>")
+		else
+			to_chat(owner, "<span class='cultitalic'><b><u>You cannot store more than [RUNELESS_MAX_BLOODCHARGE] spells without an empowering rune! Pick a spell to remove.</b></u></span>")
+		remove_spell()
+		return
+	var/entered_spell_name
+	var/datum/action/innate/cult/blood_spell/BS
+	var/list/possible_spells = list()
+	for(var/I in subtypesof(/datum/action/innate/cult/blood_spell))
+		var/datum/action/innate/cult/blood_spell/J = I
+		var/cult_name = initial(J.name)
+		possible_spells[cult_name] = J
+	possible_spells += "(REMOVE SPELL)"
+	entered_spell_name = input(owner, "Pick a blood spell to prepare...", "Spell Choices") as null|anything in possible_spells
+	if(entered_spell_name == "(REMOVE SPELL)")
+		remove_spell()
+		return
+	BS = possible_spells[entered_spell_name]
+	if(QDELETED(src) || owner.incapacitated() || !BS || (rune && !(locate(/obj/effect/rune/empower) in range(1, owner))) || (spells.len >= limit))
+		return
+	to_chat(owner,"<span class='warning'>You begin to carve unnatural symbols into your flesh!</span>")
+	SEND_SOUND(owner, sound('sound/weapons/slice.ogg',0,1,10))
+	if(!channeling)
+		channeling = TRUE
+	else
+		to_chat(owner, "<span class='cultitalic'>You are already invoking blood magic!</span>")
+		return
+	if(do_after(owner, 100 - rune * 60, target = owner))
+		if(ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			if(H.dna && NO_BLOOD in H.dna.species.species_traits)
+				var/dam_zone = pick("l_arm", "l_hand", "r_arm", "r_hand")
```

Comment:
okay can we add more then just arms and hands to pick? Only reason is that if someone is REALLY paying attnetion they are gonna get a conclusion they are a cult right off rather then just some dumbass being clumsy.

Reply:
wait am i reading this right, it damages a limb if no_blood but bleeds otherwise? You also may need to take into account mech limbs

Reply:
someone paying attention and noticing arm wounds is a feature. you cant even see wound location on examine anyway.
and yes, cultists without blood will deal damage to themselves instead of bleeding